frontend/gpay_agent/lib/agent_datamodels/agent.dart

```diff
@@ -31,4 +31,11 @@ class Agent{
         + 'AgentLatitude: $AgentLatitude, '
         + 'AgentLongitude: $AgentLongitude}';
   }
+
+  String getName() {
+    if (AgentMidName != "")
```

Comment:
Is this the default stored in db? Might want to check against this and null. Look for functions like notNullorEmpty

Reply:
Changed to ((AgentName.midName).isEmpty) 
This value cannot be null because this comes from the json string response of post request made to server. Response is of this form [
![Screenshot 2020-06-08 at 1 50 31 PM](https://user-images.githubusercontent.com/66012036/84772457-941e8600-aff8-11ea-8373-7f43bd83cc89.png)
](url)
If middle name is null, like you mentioned, it is handled when making SQL query using spanner API. This happens in java server functions. 
It the response fails entirely, this check if (response.statusCode == 200) fails in /frontend/gpay_agent/lib/agent_views/login_view.dart

Reply:
Shall I resolve this?